lib/rubygems/specification_policy.rb

```diff
@@ -417,6 +420,15 @@ def validate_removed_attributes # :nodoc:
     end
   end
 
+  def validate_extensions # :nodoc:
+    builder = Gem::Ext::Builder.new(@specification)
+
+    rake_extension = @specification.extensions.any? {|s| builder.builder_for(s) == Gem::Ext::RakeBuilder }
```

Comment:
This can be moved to Gem`::Ext::Builder` as well and we can just call `builder.rake_extension?` in here. WDYT?

Reply:
Looks good to me either as is or moving the method, whatever you prefer.

Reply:
ok, let's keep it simple for now as is